src/NuGet.Core/NuGet.Packaging/Rules/NoRefOrLibFolderInPackageRule.cs

```diff
@@ -0,0 +1,128 @@
+        {
+            var managedCodeConventions = new ManagedCodeConventions(new RuntimeGraph());
+            var collection = new ContentItemCollection();
+            collection.Load(files);
+
+            var libItems = GetContentForPattern(collection, managedCodeConventions.Patterns.CompileLibAssemblies);
+            var refItems = GetContentForPattern(collection, managedCodeConventions.Patterns.CompileRefAssemblies);
+            var buildItems = GetContentForPattern(collection, managedCodeConventions.Patterns.MSBuildFiles);
+
+            var libFrameworks = GetGroupFrameworks(libItems).ToArray();
+            var refFrameworks = GetGroupFrameworks(refItems).ToArray();
+            var buildFrameworks = GetGroupFrameworks(buildItems).ToArray();
+
+            if (libFrameworks.Length == 0 && refFrameworks.Length == 0)
+            {
+                //if you can't find the ref and lib folder, then find the build folder
+                if (buildFrameworks.Length != 0)
+                {
+                    //if you can find any folders other than native or any, raise an NU5127
+                    if (buildFrameworks.Any(t => (FrameworkConstants.DotNetAll.Satisfies(t) || IsValidFramework(t))
+                        && t.GetShortFolderName() != "dotnet"
+                        && t.GetShortFolderName() != "native"))
+                    {
+                        var possibleFrameworks = buildFrameworks.
+                            Where(t => t.IsSpecificFramework && t.GetShortFolderName() != "dotnet" && t.GetShortFolderName() != "native").
+                            Select(t => t.GetShortFolderName()).ToArray();
+
+                        (var tfmNames, var suggestedDirectories) = GenerateWarningString(possibleFrameworks);
+                        
+                        var issue = new List<PackagingLogMessage>();
+                        issue.Add(PackagingLogMessage.CreateWarning(string.Format(MessageFormat, tfmNames, suggestedDirectories),
+                            NuGetLogCode.NU5127));
+                        return issue;
+                    }
+                }
+            }
+
+            return Array.Empty<PackagingLogMessage>();
+        }
+
+        private (string, string) GenerateWarningString(string[] possibleFrameworks)
+        {
+            var copy = new string[possibleFrameworks.Length - 1];
+            Array.Copy(possibleFrameworks, copy, possibleFrameworks.Length - 1);
+
+            string tfmNames = possibleFrameworks.Length > 1 ? string.Join(", ", copy)
+                + ", and " + possibleFrameworks[possibleFrameworks.Length - 1] : possibleFrameworks[0];
+            string suggestedDirectories = possibleFrameworks.Length > 1 ?
+                CreateDirectories(possibleFrameworks) :
+                string.Format("-lib/{0}/_._", possibleFrameworks[0]);
+            
+            return (tfmNames, suggestedDirectories);
+
```

Comment:
minor: no empty line between the last statement of a method and the closing brace.